### PR TITLE
refactor(example-app): use router store in RouterEffects

### PR DESCRIPTION
**Pull request by markostanimirovic**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Route data is extracted manually from `ActivatedRoute` in `updateTitle$` effect.

## What is the new behavior?

`updateTitle$` effect uses `selectRouteData` selector.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/example-app/src/app/core/effects/router.effects.ts [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
-        map(() => {
-          let route = this.activatedRoute;
-          while (route.firstChild) route = route.firstChild;
-          return route;
-        }),
-        mergeMap((route) => route.data),
-        map((data) => `Book Collection - ${data['title']}`),
+      this.actions$.pipe(
+        ofType(routerNavigatedAction),
+        withLatestFrom(this.store.select(fromRoot.selectRouteData)),
```
Comment (timdeschryver) [inline projects/example-app/src/app/core/effects/router.effects.ts:18]: Should this be a lazy `withLatestFrom` ?

**Review by timdeschryver — COMMENTED** [review]
LGTM, but if I think we created this effect with the idea of having an example effect that doesn't use `Actions` as the source.

**Comment by markostanimirovic** [comment]
> LGTM, but if I think we created this effect with the idea of having an example effect that doesn't use `Actions` as the source.

@timdeschryver 
Okay, I can close this PR, or keep `router` as a source instead of `actions$`, but with `selectRouteData` selector, because example with router selectors also misses. What do you think?

**Comment by timdeschryver** [comment]
@markostanimirovic you can ignore that comment, we also have [another effect that doesn't use `Actions`](https://github.com/ngrx/platform/blob/master/projects/example-app/src/app/core/effects/user.effects.ts#L15). 
You can leave it as is 👍 ,  I do think you should make that read lazy tho

```ts
        concatMap(action => of(action).pipe(
          withLatestFrom(this.store.select(fromRoot.selectRouteData))
        )),
```

**Review by timdeschryver — APPROVED** [review]
LGTM

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**